apps/server/src/modules/tool/controller/api-test/tool-context.api.spec.ts

```diff
@@ -0,0 +1,170 @@
+import { EntityManager, MikroORM } from '@mikro-orm/core';
+import { ExecutionContext, INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { ContextExternalTool, Course, Permission, Role, RoleName, SchoolExternalTool, User } from '@shared/domain';
+import {
+	courseFactory,
+	mapUserToCurrentUser,
+	roleFactory,
+	schoolExternalToolFactory,
+	userFactory,
+} from '@shared/testing';
+import { ICurrentUser } from '@src/modules/authentication';
+import { JwtAuthGuard } from '@src/modules/authentication/guard/jwt-auth.guard';
+import { ServerTestModule } from '@src/modules/server/server.module';
+import { ObjectId } from 'bson';
+import { Request } from 'express';
+import request, { Response } from 'supertest';
+import { ContextExternalToolPostParams, ContextExternalToolResponse } from '../dto';
+
+import { ToolContextType } from '../../interface';
+
+describe('ToolContextController (API)', () => {
+	let app: INestApplication;
+	let em: EntityManager;
+	let orm: MikroORM;
+
+	let currentUser: ICurrentUser;
+
+	const basePath = '/tools/context';
+
+	beforeAll(async () => {
+		const moduleRef: TestingModule = await Test.createTestingModule({
+			imports: [ServerTestModule],
+		})
+			.overrideGuard(JwtAuthGuard)
+			.useValue({
+				canActivate(context: ExecutionContext) {
+					const req: Request = context.switchToHttp().getRequest();
+					req.user = currentUser;
+					return true;
+				},
+			})
+			.compile();
+
+		app = moduleRef.createNestApplication();
+		await app.init();
+		em = app.get(EntityManager);
+		orm = app.get(MikroORM);
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	afterEach(async () => {
+		await orm.getSchemaGenerator().clearDatabase();
+	});
+
+	describe('[POST] tools/context', () => {
+		describe('when creation of contextExternalTool is successfully', () => {
+			const setup = async () => {
+				const teacherRole: Role = roleFactory.build({
+					name: RoleName.TEACHER,
+					permissions: [Permission.CONTEXT_TOOL_ADMIN],
+				});
+
+				const teacher: User = userFactory.buildWithId({ roles: [teacherRole] });
+
+				const course: Course = courseFactory.buildWithId({ teachers: [teacher] });
+
+				const paramEntry = { name: 'name', value: 'value' };
+				const schoolExternalTool: SchoolExternalTool = schoolExternalToolFactory.buildWithId({
+					schoolParameters: [paramEntry],
+					toolVersion: 1,
+				});
+
+				const postParams: ContextExternalToolPostParams = {
+					schoolToolId: schoolExternalTool.id,
+					contextId: course.id,
+					contextToolName: course.name,
+					contextType: ToolContextType.COURSE,
+					parameters: [paramEntry],
+					toolVersion: 1,
+				};
+
+				em.persist([teacherRole, course, teacher, schoolExternalTool]);
```

Comment:
you can use persistAndFlush to save 1 line :)

Reply:
thanks :)